resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -254,3 +251,17 @@ private[spark] class KubernetesClientApplication extends SparkApplication {
     }
   }
 }
+
+private[spark] object KubernetesClientApplication {
+
+  def getAppName(conf: SparkConf): String = conf.getOption("spark.app.name").getOrElse("spark")
+
+  def getResourceNamePrefix(appName: String): String = {
+    val launchTime = System.currentTimeMillis()
+    s"$appName-$launchTime"
+      .toLowerCase
+      .replaceAll(" +", " ")
+      .replaceAll("\\s", "-")
+      .replaceAll("[^A-Za-z0-9\\-]", "")
```

Comment:
Might be a bit strict but if people want weird names then they should know k8s does not accept it and we use the appname for their convenience when they list pods.

Reply:
K8s resource names generally are all lower cases, and can have `-` and `.`. So you probably can also eliminate `A-Z` here and add `.`.

Reply:
Correct will update.